src/Command/Check.hs

```diff
@@ -0,0 +1,71 @@
+{-# LANGUAGE FlexibleContexts, OverloadedStrings #-}
+module Command.Check where
+
+import Data.Monoid
+import Data.List.NonEmpty(NonEmpty)
+import Options.Applicative
+import Util
+import System.IO
+import qualified Data.Text.IO as Text
+
+import qualified Backend.Target as Target
+import qualified Processor.Files as Processor
+import qualified Processor.Result as Processor
+
+data Options = Options
+  { inputFiles :: NonEmpty FilePath
+  , verbosity :: Int
+  , logFile :: Maybe FilePath
+  } deriving (Show)
+
+optionsParserInfo :: ParserInfo Options
+optionsParserInfo = info (helper <*> optionsParser)
+  $ fullDesc
+  <> progDesc "Type check a Sixten program"
+  <> header "sixten check"
+
+optionsParser :: Parser Options
+optionsParser = Options
+  <$> nonEmptySome (strArgument
```

Comment:
Could this code be shared with the `compile` options parser (see e.g. how the `run` command reuses the `compile` parser)?

Reply:
Yes, agreed. I'll push the relevant change for this 👍